Added empty state messages

morph1nX opened the pull request:
Fixes Issue #6 
Please see if everything looks and functions as expected. The code structure might need some more help. Working on it.

ayoung19 commented on popup/components/EntryList.tsx:
```diff
@@ -106,15 +108,55 @@ export const EntryList = ({ entries }: Props) => {
-      </FixedSizeList>
+      {entries.length === 0 ? (
+        <Box
+          style={{
+            display: "flex",
+            flexDirection: "column",
+            height: 450,
+            width: 700,
+            justifyContent: "center",
+            alignItems: "center",
```
Also there are other components you could use here instead of Box if you'd like to be consistent. They are `Flex`, `Stack`, and `Group`.

ayoung19 reviewed (commented):
Looks good so far. Main change is moving all the page logic in `EntryList` up to the respective `*Page` components.

morph1nX commented:
I moved the rendering logic to the respective `page` components and removed the inline styles. However I am not using the `sx` prop to style as the `Stack` component provides props like `h`, `w`, `justify` and `align` which is enough for me. Also I see that the component responds to theme changes anyways. What do you think? I have never used Mantine before lol.

ayoung19 commented:
No worries! It always takes time getting used to new libraries. I like Mantine because it's extremely customizable so it'll be very helpful if we do a redesign in the future.

Also using those props is fine, it's equivalent to using `sx` I think.

ayoung19 commented on popup/pages/AllPage.tsx:
```diff
@@ -8,8 +9,32 @@ export const AllPage = () => {
 
+  const renderNoEntriesOverlay = () => {
+    return (
+      <Stack h={450} w={700} justify="center" align="center">
```
nit: Make a `Box` that wraps around `noEntriesOverlay` in `EntryList` so the caller doesn't have to worry about sizing and remove the `h` and `w` props here.

morph1nX replied:
So you want only the text to be passed from the` page` components and the rest should be handled in the `EntryList`? Also, I should use Box instead of Stack?

ayoung19 replied:
Oh no. I'm saying `renderNoEntriesOverlay` should just be a new component that takes in `title` and `subtitle` which `AllPage` and `FavoritesPage` can share. Keep using the stack, but move only the height/width props to a `Box` component in `EntryList` that wraps the render.

ayoung19 replied:
Feel free to message me on Discord if it still doesn't make sense. IMO that approach is more maintainable (as it's in line with how other component libraries do things).

ayoung19 commented on popup/pages/AllPage.tsx:
```diff
@@ -8,8 +9,32 @@ export const AllPage = () => {
   const search = useAtomValue(searchAtom);
   const entryIdToTags = useAtomValue(entryIdToTagsAtom);
 
+  const renderNoEntriesOverlay = () => {
```
nit: Since it seems pretty similar I think we should just make this a component. Maybe something of the form like:

```
<NoEntriesOverlay title={string} subtitle={string} />
```

This is inspired by MUI's `DataGrid` slot by the way: https://mui.com/x/api/data-grid/data-grid/#DataGrid-css-noRowsOverlay

ayoung19 commented:
Looks good! Just a few more nits, thank you so much for working through these!

morph1nX commented:
Made `NoEntriesOverlay` its own component. I too had thought about this first but all the props passing put me off. Anyways I hope this approach works!
